src/tiered/tiered_cursor.c

```diff
@@ -1046,12 +1033,26 @@ __curtiered_next_random(WT_CURSOR *cursor)
     __cursor_novalue(cursor);
     WT_ERR(__curtiered_enter(curtiered, false));
 
-    for (;;) {
-        __curtiered_random_tier(session, curtiered, &c);
-        /*
-         * This call to next_random on the tier can potentially end in WT_NOTFOUND if the tier we
-         * picked is empty. We want to retry in that case.
-         */
+    /*
+     * Select a random item from a random tier. Keep trying different tiers until we either find
+     * something or run out of tiers. To do this we generate a random permutation of the tier
+     * cursors and iterate through them in the resulting order.
+     *
+     * TODO: We should weight this by the number of elements in each tier.
```

Comment:
I think this is fine for now. It isn't clear to me what it really means for the expected, common case where we've called flush_tier, and our new first entry, "file:name.000003.wt" is essentially empty and "tier:name" which covers  all the data in objects 00001 and 00002 gets its random cursor.

Reply:
See my response to @ddanderson. I'm moving the TODO to its own ticket since that's a better way to track outstanding work.